src/device-manager/WeaveDataManagementClient.cpp

```diff
@@ -1128,13 +1355,29 @@ WEAVE_ERROR WdmClient::Init(WeaveMessageLayer * apMsgLayer, Binding * apBinding)
     mFailedFlushPathStatus.clear();
     mFailedPaths.clear();
 
+    mEventStrBuffer.clear();
+    mLimitEventFetchTimeout = false;
+    memset(mLastObservedEventByImportance, 0, sizeof mLastObservedEventByImportance);
+
+    mSubscriptionTLE = false;
+
 exit:
     return WEAVE_NO_ERROR;
 }
 
 void WdmClient::SetNodeId(uint64_t aNodeId)
 {
     mSinkCatalog.SetNodeId(aNodeId);
+
+    mWdmEventProcessor.reset(new WdmEventProcessor(aNodeId, this));
```

Comment:
why we need reset event processor in set node id?

Reply:
`mWdmEventProcessor` is a unique_ptr, and WdmEventProcessor needs nodeid, here we create a new WdmEventProcessor (and destruct the old one if we need) with the node id set.